modules/service/src/main/scala/lucuma/odb/graphql/input/CoordinatesInput.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2016-2022 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.odb.graphql
+package input
+
+import cats.data.ValidatedNec
+import cats.syntax.option.*
+import cats.syntax.parallel.*
+import cats.syntax.validated.*
+import edu.gemini.grackle.Result
+import lucuma.core.math.Coordinates
+import lucuma.core.math.Declination
+import lucuma.core.math.RightAscension
+import lucuma.odb.graphql.util.Bindings.Matcher
+import lucuma.odb.graphql.util.Bindings.ObjectFieldsBinding
+
+//# Absolute coordinates relative base epoch
+//input CoordinatesInput {
+//  ra: RightAscensionInput
+//  dec: DeclinationInput
+//}
+
+final case class CoordinatesInput(
+  ra:  Option[RightAscension],
+  dec: Option[Declination]
+) {
+
+  def create: ValidatedNec[String, Option[Coordinates]] =
+    (ra, dec) match {
+      case (Some(r), Some(d)) => Coordinates(r, d).some.validNec
+      case (None, None)       => none.validNec
+      case _                  => CoordinatesInput.messages.BothRaAndDecNeeded.invalidNec
+    }
```

Comment:
Potentially controversial, but `ValidatedNec` is closer to what I think we want for argument validation than `Ior`?

Reply:
GraphQL allows you to return warnings along with the result, so I have been using `Result` for all of this stuff.

Reply:
I guess it's not a huge deal either wat. Where is `toResult` defined?

Reply:
Switched to `Result` and removed the little intermediate `CreateResult` and `UpdateResult` types since I just translated to those and then back again to `Result`.  That simplified it a bit.